docs/README.md

```diff
@@ -0,0 +1,5 @@
+# Documentation
+
```

Comment:
@mukeshpanchal27 @felixarntz it would be nice to see this TOC at the top of every doc MD file so that navigation is more seamless. Just an idea.

Reply:
Thank you, @10upsimon, for the great idea. I really appreciate it. Also, a big thanks to @felixarntz for suggesting the "Back to overview" link idea, which I really liked. I have now implemented it in the commit at https://github.com/10up/plugin-check/pull/148/commits/473fee29ea07e0fcc46abcd8fb1240c7f83d63dc. Feel free to check it out and provide any feedback.